### PR TITLE
feat(TSX): add tsx

### PR DESCRIPTION
**Pull request by nattallius**

### CODE REVIEW

File: src/plugins/settings/settings.tsx [hunk 1]
```diff
@@ -22,28 +22,18 @@ export class UIPSettings extends UIPPlugin {
...
-    [...this.childNodes].forEach( (node: HTMLElement) => {
-      $settingsList.appendChild(node);
-    });
-    this.$inner.appendChild($settingsList);
-    this.$scroll && this.$inner.appendChild(this.$scroll);
+    const content = <div className="settings-list esl-scrollable-content">
+      {this.childNodes}
+      <esl-scrollbar target="::prev(.settings-list)"></esl-scrollbar>
+    </div>;
+    this.$inner.appendChild(content);
```
Comment (yadamskaya) [inline src/plugins/settings/settings.tsx:36]:
```suggestion
    const $content = <div className="settings-list esl-scrollable-content">
      <esl-scrollbar target="::prev(.settings-list)"></esl-scrollbar>
    </div>;
    [...this.childNodes].forEach((node: HTMLElement) => {
      $content.appendChild(node);
    });
    this.$inner.appendChild($content);
```

File: src/settings/slider-setting/slider-setting.tsx [hunk 2]
```diff
@@ -20,9 +22,16 @@ export class UIPSliderSetting extends UIPSetting {
   protected connectedCallback() {
     super.connectedCallback();
 
-    this.$label.append(this.$field);
-    this.append(this.$label);
-    this.append(this.$fieldValue);
+    const inner =
```
Comment (yadamskaya) [inline src/settings/slider-setting/slider-setting.tsx:25]:
```suggestion
    const $inner =
```

File: src/settings/bool-setting/bool-setting.tsx [hunk 3]
```diff
@@ -36,26 +38,21 @@ export class UIPBoolSetting extends UIPSetting {
...
+  protected get $inconsistencyMarker() {
+    return <div className="inconsistency-marker"/> as HTMLDivElement;
   }
 
   protected connectedCallback() {
     super.connectedCallback();
-    this.insertBefore(this.$label, this.firstChild);
+    this.innerHTML = '';
+
+    const inner = <label>
```
Comment (yadamskaya) [inline src/settings/bool-setting/bool-setting.tsx:51]:
```suggestion
    const $inner = <label>
```

File: src/settings/text-setting/text-setting.tsx [hunk 4]
```diff
@@ -16,22 +18,20 @@ export class UIPTextSetting extends UIPSetting {
...
   }
 
   protected connectedCallback() {
     super.connectedCallback();
+    this.innerHTML = '';
 
-    const label = document.createElement('label');
-    label.innerText = this.label;
-    label.appendChild(this.$field);
+    const inner = <label>
```
Comment (yadamskaya) [inline src/settings/text-setting/text-setting.tsx:29]:
```suggestion
    const $inner = <label>
```

**Review by yadamskaya — COMMENTED** [review]
Great work! Thanks!
One thing in addition to comments is: 
Can you please use PascalCase naming convention for tsx files?

File: src/plugins/header/header.tsx [hunk 5]
```diff
@@ -33,16 +35,13 @@ export class UIPHeader extends UIPPlugin {
...
     this.append(optionsEl);
   }
 
   /** Renders copy icon */
   protected renderCopy(): void {
-    const icon = document.createElement('button');
-    icon.title = 'copy markup';
-    icon.classList.add('copy-icon');
-    icon.append(UIPOptionIcons.copySVG);
+    const icon = <button title="copy markup" className="copy-icon">{UIPOptionIcons.copySVG.cloneNode(true) as HTMLElement}</button>;
```
Comment (Sisha3342) [inline src/plugins/header/header.tsx:44]: Do we really need to use `.cloneNode(true)`? `UIPOptionIcons.copySVG` returns new element on every call, so there shouldn't be any reasons to copy it.

File: src/plugins/settings/settings.tsx [hunk 6]
```diff
@@ -22,28 +22,20 @@ export class UIPSettings extends UIPPlugin {
...
   /** Initializes settings layout */
   protected updateInner() {
-    const $settingsList = document.createElement('div');
-    $settingsList.className = 'settings-list esl-scrollable-content';
-    [...this.childNodes].forEach( (node: HTMLElement) => {
-      $settingsList.appendChild(node);
+    const $content = <div className="settings-list esl-scrollable-content">
+      <esl-scrollbar target="::prev(.settings-list)"></esl-scrollbar>
+    </div>;
+    [...this.childNodes].forEach((node: HTMLElement) => {
```
Comment (Sisha3342) [inline src/plugins/settings/settings.tsx:35]: There is a `childNodes.forEach` callback, so we don't need to create array here.

File: src/plugins/header/options/option/option.tsx [hunk 7]
```diff
@@ -22,17 +24,16 @@ export class UIPOption extends ESLBaseElement {
...
   protected $root: UIPRoot;
   protected config: OptionConfig;
 
   /** Builds option element from {@link OptionConfig} */
-  static createEl(optionConfig: OptionConfig): UIPOption {
+  static createEl(optionConfig: OptionConfig) {
     const option = document.createElement('uip-option') as UIPOption;
     option.setAttribute('attribute', optionConfig.optionValue);
-    option.append(optionConfig.svg);
+    option.append(optionConfig.svg.cloneNode(true));
```
Comment (Sisha3342) [inline src/plugins/header/options/option/option.tsx:36]: The same as above: do we really need to use `.cloneNode` here?

**Review by Sisha3342 — APPROVED** [review]

File: src/plugins/header/header.tsx [hunk 8]
```diff
@@ -33,16 +35,13 @@ export class UIPHeader extends UIPPlugin {
 
   /** Renders {@link UIPOptions} element */
   protected renderOptions(): void {
-    const optionsEl = document.createElement(UIPOptions.is) as UIPSnippets;
+    const optionsEl = document.createElement(UIPOptions.is) as UIPOptions;
```
[the remark this answers is not in the document]
Comment (nattallius) [reply src/plugins/header/header.tsx:38]: Yes, you are right. Remove it

File: src/plugins/header/header.tsx (hunk 5 above)
Comment (nattallius) [reply src/plugins/header/header.tsx:44]: Actually... Yes) But not in the case of copy icon, but with `UIPOptions.UIPOptionsConfig` . The problem occurs when we have more than one playground on a page. When we copy link to an element from `UIPOptionIcons` to `UIPOptions.UIPOptionsConfig.svg` it behaves as a single instance. Append the second icon to the document removes previous one.

You can look it alive at our current github page https://exadel-inc.github.io/ui-playground/examples/example/multiple-playgrounds/

File: src/plugins/header/options/option/option.tsx (hunk 7 above)
Comment (nattallius) [reply src/plugins/header/options/option/option.tsx:36]: Answered above

File: src/plugins/settings/settings.tsx [hunk 9]
```diff
@@ -22,28 +22,20 @@ export class UIPSettings extends UIPPlugin {
...
-    const $settingsList = document.createElement('div');
-    $settingsList.className = 'settings-list esl-scrollable-content';
-    [...this.childNodes].forEach( (node: HTMLElement) => {
-      $settingsList.appendChild(node);
+    const $content = <div className="settings-list esl-scrollable-content">
+      <esl-scrollbar target="::prev(.settings-list)"></esl-scrollbar>
+    </div>;
+    this.childNodes.forEach((node: HTMLElement) => {
+      $content.appendChild(node);
     });
```
Comment (yadamskaya) [inline src/plugins/settings/settings.tsx:37]:
```suggestion
$content.append(...this.childNodes);
```

**Review by yadamskaya — APPROVED** [review]

**Pull request merged**